txsni/certmaps.py

```diff
@@ -0,0 +1,50 @@
+"""
+{ hostname : certificate } maps compatible with the way the
+https://dehydrated.io/ acme client stores its certificates.
+"""
+
+from txsni.only_noticed_pypi_pem_after_i_wrote_this import (
+    certificateOptionsFromPileOfPEM
+)
+
+class PerHostnameDirectoryMap(object):
+    """
+    Per-hostname subdirectories with two pem's.
+    """
+
+    def __init__(self, directoryPath, keyName='privkey.pem', fullchainName='fullchain.pem'):
+        self.directoryPath = directoryPath
+        self.keyName = keyName
+        self.chainName = fullchainName
+
+    def __getitem__(self, hostname):
+        if hostname is None:
+            hostname = b"DEFAULT"
+        hostPath = self.directoryPath.child(hostname)
+        keyPath = hostPath.child(self.keyName)
+        fullchainPath = hostPath.child(self.chainName)
+        if keyPath.isfile() and fullchainPath.isfile():
+            return certificateOptionsFromPileOfPEM(
+                keyPath.getContent() + fullchainPath.getContent()
+            )
+        else:
+            raise KeyError("no pem files for " + hostname.decode('charmap'))
+
+
+class PerHostnameFilesMap(object):
```

Comment:
Especially after I got some requests to dingoskidneys.com with a hostname I knew nothing about, because there is an old DNS entry for my IP address, I like the "separate client, a bit of configuration, and web server integration" strategy.

Reply:
It sounds like what you really want here is better error handling in txsni or txacme?